packages/editor/src/browser/editor-navigation-contribution.ts

```diff
@@ -83,7 +84,17 @@ export class EditorNavigationContribution implements Disposable, FrontendApplica
             isEnabled: () => !!this.locationStack.lastEditLocation()
         });
         this.commandRegistry.registerHandler(EditorCommands.CLEAR_EDITOR_HISTORY.id, {
-            execute: () => this.locationStack.clearHistory(),
+            execute: async () => {
+                const confirmed = await new ConfirmDialog({
+                    title: 'Clear Editor History',
+                    msg: 'Are you sure you want to clear the history of recently opened editors?',
```

Comment:
We need to localize user-facing strings:

```suggestion
                    msg: nls.localizeByDefault('Do you want to clear the history of recently opened editors?'),
```